openedx/core/djangoapps/discussions/models.py

```diff
@@ -240,6 +242,30 @@ class DiscussionsConfiguration(TimeStampedModel):
     Associates a learning context with discussion provider and configuration
     """
 
+    def user_permissions_for_course(self, course, user):
```

Comment:
This permission check code should be part of the view/API not the model. This models code should be untouched in this PR.